src/utils/api.ts

```diff
@@ -0,0 +1,101 @@
+import { Server } from "http";
+import express, { Request, Response, Router } from "express";
+import client from "prom-client";
+import { Express } from "express-serve-static-core";
+import { getLogger } from "./logging";
+import { DBService } from "./db";
+import { Registry } from "../types";
+import { MetricsService } from "./metrics";
+
+export class ApiService {
+  protected PORT: number;
+  protected app: Express;
+  protected server: Server | null = null;
+
+  constructor(port: number) {
+    this.PORT = port;
+    this.app = express();
+    this.bootstrap();
+  }
+
+  private bootstrap() {
+    const collectDefaultMetrics = client.collectDefaultMetrics;
+    collectDefaultMetrics();
+
+    this.app.use(express.json());
+    this.app.use(express.urlencoded({ extended: true }));
+    this.app.get("/", (_req: Request, res: Response) => {
+      res.send("Moooo!");
+      MetricsService.apiRequestCounter.labels("GET", "/", "200").inc();
+    });
+    this.app.use("/api", router);
+    this.app.get("/metrics", async (_req, res) => {
+      res.set("Content-Type", client.register.contentType);
+      const response = await client.register.metrics();
+      res.send(response);
+    });
+  }
+
+  get getApp(): Express {
+    return this.app;
+  }
+
+  async start(): Promise<Server> {
+    return await new Promise((resolve, reject) => {
+      try {
+        const log = getLogger("api");
```

Comment:
should we move this log to at the beginging of the file?

Reply:
This is not possible now due to the initialization requirements introduced with `initLogging`.

Reply:
yep, i see. not a big deal then